src/components/include/utils/threads/thread.h

```diff
@@ -104,54 +104,45 @@ class Thread : public QObject {
 #else
 class Thread {
 #endif
+
+  typedef enum ThreadCommandEnum {
+    kThreadCommandNone,  // must be first
+    kThreadCommandRun,
+    kThreadCommandStop,
+    kThreadCommandFinalize  // keep last
+    // in case of new commands - update/check threadFunc()
+  } ThreadCommand;
+
+  typedef enum ThreadStateEnum {
+    kThreadStateNone,
```

Comment:
@LuxoftAKutsan, it is related to thread state, so it means that there is no thread at all.